data-entry/dataEntry.app.js

```diff
@@ -78,12 +78,62 @@
                         if (key.simple) {
                             (function(key) {
                                 key.getDomainValues().then(function success(values) {
-                                    dataEntryModel.domainValues[key.colset.columns[0].name] = [];
-                                    var domainValues = dataEntryModel.domainValues[key.colset.columns[0].name];
-
-                                    angular.forEach(values.data, function(value) {
-                                        var field = Object.keys(value)[0];
-                                        domainValues.push(value[field]);
+                                    var domainValues = dataEntryModel.domainValues[key.colset.columns[0].name] = [];
+                                    var table = key.key._table;
+                                    var annotations = table.annotations.all();
+                                    var termColumnName = null;
+
+
+                                    // TODO need to figure out how to handle if mutiple annotations are present
+                                    angular.forEach(annotations, function(annotation) {
```

Comment:
First there's no need to loop through all of the annotations. You only need to check for `tag:misd.isi.edu,2015:vocabulary`. Use the public api provided: `table.annotations.get("tag:misd.isi.edu,2015:vocabulary")`, rather than getting the `table.annotations.all()` and looping over them.